android/source/src/main/kotlin/com/gu/source/components/pager/PagerProgressButtons.kt

```diff
@@ -0,0 +1,128 @@
+    buttonColours: ButtonColours = DefaultButtonColours,
+    prevButtonContentDescription: String?,
+    nextButtonContentDescription: String?,
+    modifier: Modifier = Modifier,
+    disabledButtonColours: ButtonColours? = disabledModeButtonColours(buttonColours),
+    pageSlideAnimationSpec: AnimationSpec<Float> = DefaultPageSlideAnimationSpec,
+) {
+    val coroutineScope = rememberCoroutineScope()
+
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(8.dp),
+    ) {
+        SourceBaseIconButton(
+            size = SourceButton.Size.Small,
+            buttonColours = buttonColours,
+            disabledButtonColours = disabledButtonColours,
+            onClick = {
+                handlePageTransition(
+                    coroutineScope,
+                    pagerState,
+                    ProgressDirection.Previous,
+                    pageSlideAnimationSpec
+                )
+            },
+            enabled = pagerState.canScrollBackward,
+            modifier = Modifier.offset(x = ProgressButtonTouchAdjustment * 2),
+        ) {
+            Icon(
+                imageVector = Source.Icons.Base.ChevronLeft,
+                contentDescription = prevButtonContentDescription,
+                modifier = it,
+            )
+        }
+        SourceBaseIconButton(
+            size = SourceButton.Size.Small,
+            buttonColours = buttonColours,
+            disabledButtonColours = disabledButtonColours,
+            onClick = {
+                handlePageTransition(
+                    coroutineScope,
+                    pagerState,
+                    ProgressDirection.Next,
+                    pageSlideAnimationSpec
+                )
+            },
+            enabled = pagerState.canScrollForward,
+        ) {
+            Icon(
+                imageVector = Source.Icons.Base.ChevronRight,
+                contentDescription = nextButtonContentDescription,
+                modifier = it,
+            )
+        }
+    }
+}
+
+fun handlePageTransition(
+    coroutineScope: CoroutineScope,
+    pagerState: PagerState,
+    direction: ProgressDirection,
+    pageSlideAnimationSpec: AnimationSpec<Float>
+) {
+    coroutineScope.launch {
+        val page = when (direction) {
+            ProgressDirection.Previous -> (pagerState.currentPage - 1).coerceAtLeast(0)
+            ProgressDirection.Next -> (pagerState.currentPage + 1).coerceAtMost(pagerState.pageCount - 1)
+        }
+        pagerState.animateScrollToPage(page, animationSpec = pageSlideAnimationSpec)
+    }
+}
+
+@Composable
+@PreviewPhoneBothMode
+@PreviewTabletBothMode
+fun ProgressButtonsPreview() {
+    val pagerState = rememberPagerState(0) { 10 }
+    AppColourMode {
+        Row( modifier = Modifier
+            .background(
+                AppColour(Source.Palette.Neutral100, Source.Palette.Neutral10).current,
+            )
+            .width(600.dp)
+        ) {
+            ProgressButtons(
```

Comment:
The only thing missing here is a Pager to attach to the state so it can start emitting.

```suggestion
        Box(
            modifier = Modifier.background(
                AppColour(Source.Palette.Neutral100, Source.Palette.Neutral10).current,
            ),
        ) {
            HorizontalPager(state = pagerState) {}
            ProgressButtons(
```